sycl/include/sycl/detail/pi.hpp

```diff
@@ -1,276 +0,0 @@
-//==---------- pi.hpp - Plugin Interface for SYCL RT -----------------------==//
-//
-// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
-// See https://llvm.org/LICENSE.txt for license information.
-// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
-//
-//===----------------------------------------------------------------------===//
-
-/// \file pi.hpp
-/// C++ wrapper of extern "C" PI interfaces
-///
-/// \ingroup sycl_pi
-
-#pragma once
```

Comment:
This file was being included in at least one external software, the open-source [oneMKL Interfaces](https://github.com/oneapi-src/oneMKL/blob/f4d6b1994eb110587ced0fc9777bac7550498874/src/blas/backends/cublas/cublas_scope_handle.hpp#L31), which breaks after removal of this file. Can someone please comment why this PR is not marked as a breaking change unlike some others? Also, it would help to know what the fix would be to [oneMKL Interfaces here](https://github.com/oneapi-src/oneMKL/blob/f4d6b1994eb110587ced0fc9777bac7550498874/src/blas/backends/cublas/cublas_scope_handle.hpp#L31) to prevent errors like:
```
/export/users/gchoudha/oneMKL-interfaces/src/blas/backends/cublas/cublas_scope_handle.hpp:31:10: fatal error: 'sycl/detail/pi.hpp' file not found
   31 | #include <sycl/detail/pi.hpp>
      |          ^~~~~~~~~~~~~~~~~~~~
1 error generated.
```

Reply:
`pi.hpp` is an undocumented header which is *not* a part of our API and we do not guarantee stability of internal APIs.